rest_api/simple_supply_rest_api/messaging.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+from sawtooth_rest_api.messaging import Connection
+from sawtooth_rest_api.protobuf import client_batch_submit_pb2
+from sawtooth_rest_api.protobuf import validator_pb2
+
+from sawtooth_signing import create_context
+from sawtooth_signing import CryptoFactory
+from sawtooth_signing import secp256k1
+
+from simple_supply_rest_api.errors import ApiBadRequest
+from simple_supply_rest_api.errors import ApiInternalError
+
+from simple_supply_rest_api.transaction_creation import \
+    make_create_agent_transaction
+
+
+class Messenger(object):
+    def __init__(self, validator_url):
+        self._connection = Connection(validator_url)
+        self._context = create_context('secp256k1')
+        self._crypto_factory = CryptoFactory(self._context)
+        self._batch_signer = self._crypto_factory.new_signer(
+            self._context.new_random_private_key())
+
+    def open_validator_connection(self):
+        self._connection.open()
+
+    def close_validator_connection(self):
```

Comment:
What happens if this tries to close without having been opened?

Reply:
I believe any issues are handled in the Connection class: https://github.com/hyperledger/sawtooth-core/blob/master/rest_api/sawtooth_rest_api/messaging.py#L326